### ISSUE TITLE
consensus: non-responsive to CTRL-C if consensus state panics

### ISSUE THREAD

**Issue by ebuchman** [tendermint/tendermint#2072]
If we panic in the consensus, we're not responsive to CTRL-C

**Comment by cwgoes** [comment]
ACK, can replicate.

**Comment by melekes** [comment]
```
SIGABRT: abort
PC=0x45b2c1 m=0 sigcode=0

goroutine 0 [idle]:
runtime.futex(0x149b808, 0x0, 0x0, 0x0, 0x0, 0x4288fc, 0x0, 0x0, 0x7ffc7e3a66b0, 0x40f1db, ...)
        /usr/local/go/src/runtime/sys_linux_amd64.s:530 +0x21
runtime.futexsleep(0x149b808, 0x0, 0xffffffffffffffff)
        /usr/local/go/src/runtime/os_linux.go:45 +0x4b
runtime.notesleep(0x149b808)
        /usr/local/go/src/runtime/lock_futex.go:151 +0x9b
runtime.stopm()
        /usr/local/go/src/runtime/proc.go:1952 +0xe5
runtime.startlockedm(0xc420476300)
        /usr/local/go/src/runtime/proc.go:2131 +0x99
runtime.schedule()
        /usr/local/go/src/runtime/proc.go:2554 +0x6f
runtime.park_m(0xc420000180)
        /usr/local/go/src/runtime/proc.go:2604 +0xb6
runtime.mcall(0x0)
        /usr/local/go/src/runtime/asm_amd64.s:351 +0x5b

goroutine 1 [select (no cases)]:
github.com/tendermint/tendermint/libs/common.TrapSignal(0xc4207e93a0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/common/os.go:52 +0xd7
github.com/tendermint/tendermint/node.(*Node).RunForever(0xc420098900)
        /home/vagrant/go/src/github.com/tendermint/tendermint/node/node.go:517 +0x5c
github.com/tendermint/tendermint/cmd/tendermint/commands.NewRunNodeCmd.func1(0xc4200beb40, 0xc420078f80, 0x1, 0x4, 0x0, 0x0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/cmd/tendermint/commands/run_node.go:64 +0x2be
github.com/tendermint/tendermint/vendor/github.com/spf13/cobra.(*Command).execute(0xc4200beb40, 0xc420078f40, 0x4, 0x4, 0xc4200beb40, 0xc420078f40)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/spf13/cobra/command.go:698 +0x46d
github.com/tendermint/tendermint/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x148e840, 0xce4ca0, 0x12, 0x20)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/spf13/cobra/command.go:783 +0x2e4
github.com/tendermint/tendermint/vendor/github.com/spf13/cobra.(*Command).Execute(0x148e840, 0x20, 0xc420791e98)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/spf13/cobra/command.go:736 +0x2b
github.com/tendermint/tendermint/libs/cli.Executor.Execute(0x148e840, 0xf35148, 0x2, 0xc42079acc0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/cli/setup.go:89 +0x4e
main.main()
        /home/vagrant/go/src/github.com/tendermint/tendermint/cmd/tendermint/main.go:45 +0x24d

goroutine 5 [syscall]:
os/signal.signal_recv(0xfa4e00)
        /usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
        /usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 22 [select, locked to thread]:
runtime.gopark(0xf35508, 0x0, 0xdd21cf, 0x6, 0x18, 0x1)
        /usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc42004ef50, 0xc420198480)
        /usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
        /usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 7 [chan receive]:
github.com/tendermint/tendermint/libs/autofile.initSighupWatcher.func1(0xc4201720c0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/sighup_watcher.go:25 +0x67
created by github.com/tendermint/tendermint/libs/autofile.initSighupWatcher
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/sighup_watcher.go:24 +0x10e

goroutine 23 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc42015c2a0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:206 +0x152
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:237 +0x171

goroutine 24 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc4201cf860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:90 +0xff
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:142 +0x49d

goroutine 25 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc4201cf860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_state.go:101 +0x127
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:143 +0x4bf

goroutine 26 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc4201cf860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:834 +0x369
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:149 +0x627

goroutine 27 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc4201cf860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:762 +0x15d
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:150 +0x649

goroutine 28 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc42015c460)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:206 +0x152
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:237 +0x171

goroutine 29 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc42009a820)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:90 +0xff
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:142 +0x49d

goroutine 30 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc42009a820)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_state.go:101 +0x127
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:143 +0x4bf

goroutine 31 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc42009a820)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:834 +0x369
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:149 +0x627

goroutine 32 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc42009a820)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:762 +0x15d
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:150 +0x649

goroutine 12 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc4201ceea0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:834 +0x369
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:149 +0x627

goroutine 37 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc42015c380)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:206 +0x152
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:237 +0x171

goroutine 36 [chan receive]:
github.com/tendermint/tendermint/libs/autofile.(*AutoFile).processTicks(0xc42017cec0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/autofile.go:68 +0x5d
created by github.com/tendermint/tendermint/libs/autofile.OpenAutoFile
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/autofile.go:54 +0x189

goroutine 13 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc4201ceea0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:762 +0x15d
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:150 +0x649

goroutine 38 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc4201ceea0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:90 +0xff
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:142 +0x49d

goroutine 39 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc4201ceea0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_state.go:101 +0x127
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:143 +0x4bf

goroutine 14 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.(*BufferPool).drain(0xc42015c7e0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:206 +0x152
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util.NewBufferPool
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/util/buffer_pool.go:237 +0x171

goroutine 15 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).compactionError(0xc42009b860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:90 +0xff
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:142 +0x49d

goroutine 16 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain(0xc42009b860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_state.go:101 +0x127
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:143 +0x4bf

goroutine 49 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction(0xc42009b860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:834 +0x369
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:149 +0x627

goroutine 50 [select]:
github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.(*DB).mCompaction(0xc42009b860)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db_compaction.go:762 +0x15d
created by github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb.openDB
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/syndtr/goleveldb/leveldb/db.go:150 +0x649

goroutine 52 [IO wait]:
internal/poll.runtime_pollWait(0x7f5fe3276c90, 0x72, 0x0)
        /usr/local/go/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc42011d698, 0x72, 0xc42017d900, 0x0, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc42011d698, 0xffffffffffffff00, 0x0, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Accept(0xc42011d680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/internal/poll/fd_unix.go:372 +0x1a8
net.(*netFD).accept(0xc42011d680, 0xbad4c3, 0x1, 0x0)
        /usr/local/go/src/net/fd_unix.go:238 +0x42
net.(*TCPListener).accept(0xc42000c4f0, 0xbad4e5, 0x2, 0x0)
        /usr/local/go/src/net/tcpsock_posix.go:136 +0x2e
net.(*TCPListener).Accept(0xc42000c4f0, 0x1, 0xd42600, 0xc42080ce01, 0xc42209f560)
        /usr/local/go/src/net/tcpsock.go:259 +0x49
github.com/tendermint/tendermint/vendor/golang.org/x/net/netutil.(*limitListener).Accept(0xc42209eb10, 0xc420032080, 0xca91e0, 0x1481a40, 0xdb1540)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/golang.org/x/net/netutil/listen.go:48 +0x4e
net/http.(*Server).Serve(0xc42079d450, 0xfa89c0, 0xc42209eb10, 0x0, 0x0)
        /usr/local/go/src/net/http/server.go:2770 +0x1a5
net/http.Serve(0xfa89c0, 0xc42209eb10, 0xfa0be0, 0xc42209f4a0, 0xfa0be0, 0xc42209f4a0)
        /usr/local/go/src/net/http/server.go:2389 +0x73
github.com/tendermint/tendermint/rpc/lib/server.StartHTTPServer.func1(0xc4201797d0, 0xf9faa0, 0xc42209e5a0, 0xfaa600, 0xc420183560)
        /home/vagrant/go/src/github.com/tendermint/tendermint/rpc/lib/server/http_server.go:60 +0xc1
created by github.com/tendermint/tendermint/rpc/lib/server.StartHTTPServer
        /home/vagrant/go/src/github.com/tendermint/tendermint/rpc/lib/server/http_server.go:59 +0x4d5

goroutine 53 [chan receive]:
github.com/tendermint/tendermint/libs/autofile.(*AutoFile).processTicks(0xc42080a000)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/autofile.go:68 +0x5d
created by github.com/tendermint/tendermint/libs/autofile.OpenAutoFile
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/autofile.go:54 +0x189

goroutine 54 [chan receive]:
github.com/tendermint/tendermint/libs/autofile.(*Group).processTicks(0xc4201c2640)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/group.go:203 +0x6b
created by github.com/tendermint/tendermint/libs/autofile.(*Group).OnStart
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/autofile/group.go:105 +0x3f

goroutine 58 [select]:
github.com/tendermint/tendermint/p2p/pex.(*addrBook).saveRoutine(0xc42015c540)
        /home/vagrant/go/src/github.com/tendermint/tendermint/p2p/pex/addrbook.go:490 +0x14d
created by github.com/tendermint/tendermint/p2p/pex.(*addrBook).OnStart
        /home/vagrant/go/src/github.com/tendermint/tendermint/p2p/pex/addrbook.go:139 +0x81

goroutine 59 [select]:
github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).ensurePeersRoutine(0xc4200bca00)
        /home/vagrant/go/src/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:335 +0x15b
created by github.com/tendermint/tendermint/p2p/pex.(*PEXReactor).OnStart
        /home/vagrant/go/src/github.com/tendermint/tendermint/p2p/pex/pex_reactor.go:132 +0x114

goroutine 62 [chan receive]:
github.com/tendermint/tendermint/consensus.(*ConsensusState).Wait(...)
        /home/vagrant/go/src/github.com/tendermint/tendermint/consensus/state.go:325
github.com/tendermint/tendermint/consensus.(*ConsensusReactor).OnStop(0xc42011cc80)
        /home/vagrant/go/src/github.com/tendermint/tendermint/consensus/reactor.go:80 +0x82
github.com/tendermint/tendermint/libs/common.(*BaseService).Stop(0xc42011cc80, 0x0, 0x0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/common/service.go:152 +0x1c7
github.com/tendermint/tendermint/p2p.(*Switch).OnStop(0xc42009b040)
        /home/vagrant/go/src/github.com/tendermint/tendermint/p2p/switch.go:225 +0x1e8
github.com/tendermint/tendermint/libs/common.(*BaseService).Stop(0xc42009b040, 0x0, 0x0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/common/service.go:152 +0x1c7
github.com/tendermint/tendermint/node.(*Node).OnStop(0xc420098900)
        /home/vagrant/go/src/github.com/tendermint/tendermint/node/node.go:490 +0xb6
github.com/tendermint/tendermint/libs/common.(*BaseService).Stop(0xc420098900, 0x0, 0x0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/common/service.go:152 +0x1c7
github.com/tendermint/tendermint/node.(*Node).RunForever.func1()
        /home/vagrant/go/src/github.com/tendermint/tendermint/node/node.go:518 +0x2c
github.com/tendermint/tendermint/libs/common.TrapSignal.func1(0xc42081a720, 0xc4207e93a0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/common/os.go:47 +0xcb
created by github.com/tendermint/tendermint/libs/common.TrapSignal
        /home/vagrant/go/src/github.com/tendermint/tendermint/libs/common/os.go:43 +0xd2

rax    0xca
rbx    0x149b6c0
rcx    0x45b2c3
rdx    0x0
rdi    0x149b808
rsi    0x0
rbp    0x7ffc7e3a6678
rsp    0x7ffc7e3a6630
r8     0x0
r9     0x0
r10    0x0
r11    0x286
r12    0xc420198458
r13    0xea
r14    0x200
r15    0x200
rip    0x45b2c1
rflags 0x286
cs     0x33
fs     0x0
gs     0x0
```

**Comment by ebuchman** [comment]
Fixed in #2135
